core/embed/rust/src/ui/display/loader.rs

```diff
@@ -337,7 +338,7 @@ pub fn loader_rust(
 
             icon_buffer_used.buffer[icon_offset as usize..(icon_offset + icon_width / 2) as usize]
                 .copy_from_slice(
-                    &unwrap!(icon_data).toif.data[(y_i * (icon_width / 2)) as usize
+                    &unwrap!(icon_data).toif.zdata()[(y_i * (icon_width / 2)) as usize
```

Comment:
Don't we need to decompress the data here first? Are we even using loader with icon anywhere? Bootloader?

Reply:
yes there is a bug in here, the `loader_rust` is actually meant to receive already decompressed data, which allows to skip repeated decompression when updating progress. 

in bootloader the icons are used, so i made a fix there when i noticed this after rebase, but maybe you can cherry pick it here: https://github.com/trezor/trezor-firmware/commit/e8bfaaf0ce8931678427c9ef304bce3f83feda70

Reply:
Let's fix it in that PR if it's only used in bootloader.